src/aclic.adoc

```diff
@@ -252,72 +242,62 @@ Of the main ACLIC goals, these targets are addressed here:
 can be used at with the same trap handlers as one developed for AIA.
 
-At config-time, the interface differs,
-but ACLIC uses the same structure for configuration as the APLIC.
-
-=== New Interrupt Delivery Mode
-
-The Incoming Message Signaled Interrupt Controller (IMSIC) contains the {eidelivery} CSR,
-which defines the interrupt delivery mode.
-A new delivery mode is defined:
-
-[%autowidth,float="center",align="center",cols=">,<",grid=none,frame=none]
-|===
-|0 = | Interrupt delivery is disabled
-|1 = | Interrupt delivery from the interrupt file is enabled
-|0x20000000 = | Interrupt delivery from an ACLIC (new)
-|0x40000000 = | Interrupt delivery from a PLIC or APLIC is enabled
-|===
-
-If Smaclic or Ssaclic extensions are present, reset initializes {eidelivery} to 0x20000000.
+Since the ACLIC is closely coupled to a single hart for which AIA already defines an
+indirect CSR access mechanism, interrupt source and priority configuration is performed 
+using new indirect CSRs instead of the memory-mapped registers of an external APLIC. 
+The source and priority configuration data follow the same definitions as the equivalent APLIC fields. 
 
-When interrupts are delivered from an ACLIC, the following behavior is valid:
+Note: 
+  * A hart cannot have more than one closely coupled external interrupt controller; 
+  * in particular a hart cannot have both an ACLIC and an IMSIC. 
+  * A system can contain an APLIC as well as a closely coupled ACLIC; in this case {eidelivery} is 
+    used to select which interrupt controller is connected to each of the CLINT external interrupt inputs.   
 
-The IMSIC registers {eithreshold}, {eipk} and {eiek} serve the same functionality as with {eidelivery} = 1.
+An ACLIC supports one interrupt file for machine mode interrupts.
+The Ssaclic extension adds a second interrupt file for supervisor mode interrupts. 
 
-The `eip` array and `eie` array act as an alias of the pending and enable bits of the connected APLIC domain.
+The AIA-defined interrupt file configuration registers {eidelivery}, {eithreshold}, {eipk} and {eiek} work as
+described in AIA Chapter 3 "Incoming MSI Controller".  This includes their reset behaviour.
 
-The {eithreshold} CSR is an alias of the {ithreshold}.
-For the interrupt enable and interrupt pending bits {eipk} and {eiek},
-a write with the value 1 performs a write access with the value 1 to the corresponding {setip} or {setie} register,
-while a write with the value 0 performs a write access with value 1 to the corresponding {in_clrip} or {clrie} register.
-On read of {eipk} or {eiek}, these CSRs are an alias of {setip}[_k_] and {setie}[_k_], respectively.
+The number of interrupt identities supported by an ACLIC interrupt file is one less that a multiple of 64,
+unless the number of identities is less than 64. 
+Unlike the IMSIC for which the minimum number of supported identities is 63,
+an ACLIC can additionally support 7, 15, 31, or 63 interrupt identities. 

```

Comment:
Is this necessary? They could just be read-only zero. I.e. stick to the minimum of 64